### PR TITLE
UI : User profile page improvements

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes :

I worked on the user profile to avoid unnecessary API calls for active tabs.

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.tsx [hunk 1]
```diff
@@ -778,7 +784,7 @@ const Users = ({
         <div className="tw--mt-4 tw-px-1.5 tw-flex tw-justify-between">
           <div className="tw-relative">
             <AntDButton
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/Users.component.tsx:786]: let's name this as a Button, and if we are using any custom button component, this is a good chance to replace that with antd button

File: openmetadata-ui/src/main/resources/ui/src/pages/UserPage/UserPage.component.tsx [hunk 2]
```diff
@@ -54,7 +55,8 @@ import { showErrorToast } from '../../utils/ToastUtils';
 
 const UserPage = () => {
   const { t } = useTranslation();
-  const { username, tab } = useParams<{ [key: string]: string }>();
+  const { username, tab = UserProfileTab.ACTIVITY } =
+    useParams<{ [key: string]: string }>();
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/UserPage/UserPage.component.tsx:59]: Lets utilize typescript here and give proper types

File: openmetadata-ui/src/main/resources/ui/src/pages/UserPage/UserPage.component.tsx [hunk 3]
```diff
@@ -113,27 +115,29 @@ const UserPage = () => {
...
-      myDataSearchIndex
-    )
-      .then((res) => {
-        const hits = res?.data?.hits?.hits as SearchEntityHits;
+    if (userData.id) {
+      try {
+        const response = await searchData(
+          fetchOwnedEntities
+            ? `owner.id:${userData.id}`
+            : `followers:${userData.id}`,
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/UserPage/UserPage.component.tsx:128]: as improvement lets extract this condition outside of try block and give fine query to the function

File: openmetadata-ui/src/main/resources/ui/src/pages/UserPage/UserPage.component.tsx [hunk 4]
```diff
@@ -148,15 +152,15 @@ const UserPage = () => {
...
           });
         }
-      })
-      .catch((err: AxiosError) => {
+      } catch (error) {
         showErrorToast(
-          err,
+          error as AxiosError,
           t('server.entity-fetch-error', {
             entity: `${fetchOwnedEntities ? 'Owned' : 'Follwing'} Entities`,
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/pages/UserPage/UserPage.component.tsx:159]: localization

**Review by chirag-madlani — APPROVED** [review]

**Review by ShaileshParmar11 — COMMENTED** [review]
I left a couple of comments, as part of the improvement it would be great if you also take care of localization as well as remove the tailwind class for the user profile page

**Comment by Sachin-chaurasiya** [comment]
@ShaileshParmar11 , Thanks for the review.
As this PR is mostly about the API call improvements so I will take care of these comments in another patch.

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**